pkg/controllers/status/cluster_status_controller.go

```diff
@@ -111,15 +113,15 @@ func (c *ClusterStatusController) SetupWithManager(mgr controllerruntime.Manager
 }
 
 func (c *ClusterStatusController) syncClusterStatus(cluster *clusterv1alpha1.Cluster) (controllerruntime.Result, error) {
+	var currentClusterStatus = clusterv1alpha1.ClusterStatus{}
+
 	// create a ClusterClient for the given member cluster
 	clusterClient, err := c.ClusterClientSetFunc(cluster.Name, c.Client, c.ClusterClientOption)
 	if err != nil {
 		klog.Errorf("Failed to create a ClusterClient for the given member cluster: %v, err is : %v", cluster.Name, err)
-		return controllerruntime.Result{Requeue: true}, err
+		return c.setStatusCollectionFailedCondition(cluster, currentClusterStatus, fmt.Sprintf("failed to create a ClusterClient: %v", err))
```

Comment:
If the error persists, I'm afraid this will always update clusters.
I think we should update the status only when the condition really changed.

Reply:
It won't always update clusters, because we have retain the `LastTransitionTime` in `setTransitionTime` function.

Reply:
Yeah, you are right.